### ISSUE TITLE
DatePicker has incorrect type for value prop

### ISSUE THREAD

**Issue by aqeelat** [mui/mui-x#14765]
### Search keywords

datepicker type value

### Latest version

- [X] I have tested the latest version

### Affected products

Date and Time Pickers

### Steps to reproduce

Link to live example: https://stackblitz.com/edit/github-gm5jav

It is very hard to create a shareable repro, but the component could be a simple `<DatePicker value={new Date()} />`

### Current behavior

`value` only accepts `null` or `undefined`

### Expected behavior

To have the correct type

### Context

It is very hard to create a shareable repro, but the issue is that we have a monorepo with a couple of react apps, and a separate package for our common stuff such as ui elements and forms, some of which have DatePickers.

However, because we're not surrounding our shared components with a LocalizationProvider, the type for `DatePicker.value` becomes `never`. (but the packages that use them do actually have LocalizationProviders that are setup correctly)

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 15.0
  Binaries:
    Node: 21.7.3 - ~/.nvm/versions/node/v21.7.3/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v21.7.3/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: 129.0.6668.70
    Edge: Not Found
    Safari: 18.0
```
</details>

**Comment by flaviendelangle** [comment]
Hi,

You should be able to fix your problem using [this doc section](https://mui.com/x/react-date-pickers/base-concepts/#typing-of-the-date)

**Comment by flaviendelangle** [comment]
@LukasTy if this is doable, would you be in favor of typing the date to `any` when there is no date type registered for `PickerValidDate`?
That way, the fallback is `any` instead of `never` which is probably less brutal :laughing:

**Comment by aqeelat** [comment]
This solved it! Thank you.

I recommend making this doc more prominent. Maybe making typescript a top level document.

**Comment by LukasTy** [comment]
> @LukasTy if this is doable, would you be in favor of typing the date to `any` when there is no date type registered for `PickerValidDate`?
> That way, the fallback is `any` instead of `never` which is probably less brutal 😆

Oh, shoot. 🙈 
Well, that would clearly be much better for such use cases. 👍 💯

**Comment by flaviendelangle** [comment]
The main downside is that people won't discover how to actually get the correct typing.

People with a classic setup will have a correct typing like today
People with a setup that currently throw an error will just have `any` and will probably stay with `any`.

Not sure how to get the best of both worlds though...

**Comment by LukasTy** [comment]
> The main downside is that people won't discover how to actually get the correct typing.

Yes, that is not an ideal downside, but I think that fallback is still better than a case where you can hardly avoid the error.
I think that users, who care about correct typing will take the time to investigate how to apply it, but we will unblock those, who are fine with not the most ideal solution.
WDYT? 🤔 

> Not sure how to get the best of both worlds though...

I don't think we can, given how the typing is setup and the various use cases that users might have. 🤷

**Comment by flaviendelangle** [comment]
I also think `any` is better than an error where most people have to create a Github issue because it's hard to find an answer by yourself on the doc.

Maybe we could just add one line in the JSDoc of the `value` prop (not the other props) saying that "if you have `any` as the date type here, please look at this doc section.
Other than that I don't have ideas.

**Comment by LukasTy** [comment]
> Maybe we could just add one line in the JSDoc of the `value` prop (not the other props) saying that "if you have `any` as the date type here, please look at this doc section.
> Other than that I don't have ideas.

Yes, I came to the same solution. Steer users to the documentation on how to fix it. 🤔 🙈
